tests/test_chiral_utils.py

```diff
@@ -138,3 +139,51 @@ def test_find_chiral_bonds():
     mol = Chem.AddHs(Chem.MolFromSmiles(r"N(C)C(C)=O"))
     res = chiral_utils.find_chiral_bonds(mol)
     assert res == set([(0, 2)])
+
+
+def test_chiral_flip_check():
+
+    # test maps that can swap "chirality of methane"
+    mol_a = Chem.AddHs(Chem.MolFromSmiles("C"))
+    mol_b = Chem.AddHs(Chem.MolFromSmiles("C"))
+
+    AllChem.EmbedMolecule(mol_a, randomSeed=0)
+    AllChem.EmbedMolecule(mol_b, randomSeed=0)
+
+    conf_a = mol_a.GetConformer(0).GetPositions()
+    conf_b = mol_b.GetConformer(0).GetPositions()
+
+    chiral_set_a = ChiralRestrIdxSet.from_mol(mol_a, conf_a)
+    chiral_set_b = ChiralRestrIdxSet.from_mol(mol_b, conf_b)
+
+    assert len(chiral_set_a.restr_idxs) == 4
+    assert len(chiral_set_b.restr_idxs) == 4
+
+    identity_map = np.array([(i, i) for i in range(len(conf_a))])
+
+    swap_map = np.array(identity_map)
+    swap_map[1, 1] = 2
+    swap_map[2, 1] = 1
+
+    identity_flips = find_atom_map_chiral_conflicts(identity_map, chiral_set_a, chiral_set_b, mode="flip")
+    identity_undefineds = find_atom_map_chiral_conflicts(identity_map, chiral_set_a, chiral_set_b, mode="undefined")
+    assert len(identity_flips) == 0
+    assert len(identity_undefineds) == 0
+
+    swap_map_flips = find_atom_map_chiral_conflicts(swap_map, chiral_set_a, chiral_set_b, mode="flip")
+    swap_map_undefineds = find_atom_map_chiral_conflicts(swap_map, chiral_set_a, chiral_set_b, mode="undefined")
+    assert len(swap_map_flips) == 8  # TODO: deduplicate idxs?
+    assert len(swap_map_undefineds) == 0
+
+    # test maps where atom chirality is defined in one endstate, undefined in other
+    mol_b = Chem.AddHs(Chem.MolFromSmiles("N"))
+    AllChem.EmbedMolecule(mol_b, randomSeed=0)
+    conf_b = mol_b.GetConformer(0).GetPositions()
+    chiral_set_b = ChiralRestrIdxSet.from_mol(mol_b, conf_b)
+    assert len(chiral_set_b.restr_idxs) == 0
+
+    partial_map = identity_map[:4]
+    partial_map_flips = find_atom_map_chiral_conflicts(partial_map, chiral_set_a, chiral_set_b, mode="flip")
+    partial_map_undefineds = find_atom_map_chiral_conflicts(partial_map, chiral_set_a, chiral_set_b, mode="undefined")
+    assert len(partial_map_flips) == 0
+    assert len(partial_map_undefineds) > 0
```

Comment:
I would add one more mixed test with the following mapping that mixes undefined with swaps:

```
[[0,0],
[1,2],
[3,3],
[4,4]]
```
this way we have both undefined and swaps

Reply:
Good call -- would strengthen test!

Reply:
I didn't see a way to mix both of these cases using any mapping of this specific molecule pair, but added a test that mixes both cases in https://github.com/proteneer/timemachine/pull/908/commits/04ba0f0806ac5a9430444f2ea5c9a35f3d1a7c58

Reply:
your change makes sense, I think was thinking of setting up a case of sp3->sp3 but the core is only of size 4, and one of them is mismatched (eg. a down H matching an up H), so this wouldn't be "undefined" by rather "missing"